arduino-ide-extension/src/browser/contributions/first-start-installer.ts

```diff
@@ -0,0 +1,45 @@
+import { LocalStorageService } from '@theia/core/lib/browser/storage-service';
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { BoardsService, LibraryService } from '../../common/protocol';
+import { Contribution } from './contribution';
+
+const INIT_LIBS_AND_PACKAGES = 'initializedLibsAndPackages';
+
+@injectable()
+export class FirstStartInstaller extends Contribution {
+  @inject(LocalStorageService)
+  private readonly localStorageService: LocalStorageService;
+
+  @inject(BoardsService)
+  private readonly boardsService: BoardsService;
+
+  @inject(LibraryService)
+  private readonly libraryService: LibraryService;
+
+  override onReady(): void {
+    this.localStorageService
+      .getData(INIT_LIBS_AND_PACKAGES)
+      .then(async (value) => {
+        const isFirstStartup = !value;
+        if (isFirstStartup) {
+          await this.localStorageService.setData(INIT_LIBS_AND_PACKAGES, true);
+          const avrPackage = await this.boardsService.getBoardPackage({
+            id: 'arduino:avr',
+          });
+          const builtInLibrary = (
+            await this.libraryService.search({
+              query: 'Arduino_BuiltIn',
+            })
+          )[0];
+
+          !!avrPackage &&
+            (await this.boardsService.install({ item: avrPackage }));
+          !!builtInLibrary &&
+            (await this.libraryService.install({
+              item: builtInLibrary,
+              installDependencies: true,
+            }));
+        }
+      });
+  }
+}
```

Comment:
Make sure to remove this as we already put this code [into a separate contribution](https://github.com/arduino/arduino-ide/commit/73835eced3bda7657308bf582bd5e5690d73879a#diff-8e1a8da61c276800216f425dc94832341b6b2b17639b36e837399fb13102b128).